codec_specs.md

```diff
@@ -1,21 +1,61 @@
+### Codec ID
 
+Each codec supported for storage in Matroska MUST have a unique `Codec ID`. Each `Codec ID` MUST be prefixed with the string from the following table according to the associated type of the codec. All characters of a `Codec ID Prefix` MUST be capital letters (A-Z) except for the last character of a `Codec ID Prefix` which MUST be a hypen ("-").
+
+Codec Type | Codec ID Prefix
+-----------|----------------
+Video      | "V_"
+Audio      | "A_"
+Subtitle   | "S_"
+Button     | "B_"
+
+Each `Codec ID` MUST include a `Major Codec ID` immediately following the `Codec ID Prefix`. A `Major Codec ID` MAY be followed by an OPTIONAL `Codec ID Suffix` to communicate a refinement of the `Major Codec ID`. If a `Codec ID Suffix` is used, then the `Codec ID` MUST include a forward slash ("/") as a separator between the `Major Codec ID` and the `Codec ID Suffix`. The `Major Codec ID` MUST be composed of only capital letters (A-Z) and numbers (0-9). The `Codec ID Suffix` MUST be composed of only capital letters (A-Z), numbers (0-9), and formard slash ("/").
+
+The following table provides examples of valid `Codec IDs` and their components:
+
+Codec ID Prefix | Major Codec ID | Separator | Codec ID Suffix | Codec ID
+A_              | AAC            | /         | MPEG2/LC/SBR    | A_AAC/MPEG2/LC/SBR
+V_              | V_MPEG4        | /         | ISO/ASP         | V_MPEG4/ISO/ASP
+V_              | MPEG1          |           |                 | V_MPEG1
+
+### Codec Name
+
+Each encoding supported for storage in Matroska MUST have a Codec Name. The Codec Name provides a readable label for the encoding.
+
+### Description
+
+An optional description for the encoding.
+
+### Initialisation
+
+Each encoding supported for storage in Matroska MUST have a defined Initialisation. The Initialisation MUST describe the storage of data necessary to initialize the decoder, which MUST be stored within the `CodecPrivate Element`. When the Initialisation is updated within a track then that updated Initialisation data MUST be written into the `CodecState Element` of the first `Cluster` to require it. If the encoding does not require any form of Initialisation then `none` MUST be used to define the Initialisation and the `CodecPrivate Element` MUST be void.
+
+### Citation
+
+Documentation of the associated normative and informative references for the codec is RECOMMENDED.
+
+### Deprecation Date
+
+A timestamp, expressed in [@!RFC3339] that notes when support for the `Codec Mapping` within Matroska was deprecated. If a Codec is defined with a `Deprecation Date`, then it is RECOMMENDED that Matroska writers SHOULD NOT use the `Codec Mapping` after the `Deprecation Date`.
```

Comment:
I used `Codec Mapping` rather than `Codec` because I didn't want saying that `A_AAC/MPEG2/LC/SBR` was deprecated could imply that `AAC` as an encoding format was deprecated. Since it is just that particular mapping of the codec to Matroska was is deprecated and not the codec itself.